actionpack/lib/action_dispatch/routing/route_set.rb

```diff
@@ -823,7 +823,7 @@ def url_for(options, route_name = nil, url_strategy = UNKNOWN, method_name = nil
         params = route_with_params.params
 
         if options.key? :params
-          params.merge! options[:params]
+          params.merge! options[:params] if options[:params].is_a?(Hash)
```

Comment:
Should we ignore it or should it be a error?

Reply:
That's a good question. My opinion is that it should be ignored and not raise an error. 
This is something that started breaking for me as I began upgrading my application to rails 6 from 5. So I also think that this shouldn't be a breaking change from older versions.

But yea that's, like, just my opinion man.